ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -230,15 +232,18 @@ data instance BlockQuery (ShelleyBlock proto era) :: Type -> Type where
                   (Map (StakeCredential (EraCrypto era)) Coin)
 
   GetConstitution
-    :: BlockQuery (ShelleyBlock proto era) (Maybe (LC.Constitution era))
+    :: CG.ConwayEraGov era
+    => BlockQuery (ShelleyBlock proto era) (Maybe (LC.Constitution era))
 
   GetGovState
-    :: BlockQuery (ShelleyBlock proto era) (LC.GovState era)
+    :: CG.ConwayEraGov era
+    => BlockQuery (ShelleyBlock proto era) (LC.GovState era)
```

Comment:
As discussed on Slack: This is meaningful even pre-Conway.

Reply:
Done